File: C#/TollCalculator/TollCalculator/Vehicle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TollCalculator;

namespace TollCalculator
{
    public abstract class Vehicle : IVehicle
    {

        public string VehicleType => this.GetType().Name;

        public virtual bool IsTollFreeVehicle()
        {
            throw new NotImplementedException();
        }
    }
}

File: Java/app/src/main/java/calculator/vehicle/Vehicle.java
package calculator.vehicle;

public class Vehicle
{
    private VehicleType type;
    private String registrationId;

    public Vehicle(VehicleType type, String registrationId)
    {
        this.type = type;
        this.registrationId = registrationId;
    }

    public VehicleType getType()
    {
        return type;
    }

    public String getRegistrationId()
    {
        return registrationId;
    }
}
